relay.go

```diff
@@ -54,8 +57,9 @@ func (e RelayError) Error() string {
 	return fmt.Sprintf("error opening relay circuit: %s (%d)", pb.CircuitRelay_Status_name[int32(e.Code)], e.Code)
 }
 
-func NewRelay(ctx context.Context, h host.Host, opts ...RelayOpt) (*Relay, error) {
+func NewRelay(ctx context.Context, h host.Host, upgrader *tptu.Upgrader, opts ...RelayOpt) (*Relay, error) {
```

Comment:
where does one get a transport upgrader?

Reply:
We construct it when constructing the transports.

https://github.com/libp2p/go-libp2p/blob/29228f250fa6b70ad45cfef02386353a8c36c0f9/config/config.go#L105-L143